### PR TITLE
Replace `snafu` with `this_error` and remove global error enum

### PR DESCRIPTION
**Pull request by imabdulbasit**
This PR:
- Removes global error enum by moving the variants to a BlockBuildingError enum
- removes snafu dependency from everywhere except from the availability endpoint, as hotshot query service still uses it
- moves error types into their relevant modules
- use parse_duration and parse_size from espresso_types::utils

This PR does not remove snafu dependency completely, as we need to replace snafu with this_error in hotshot query service

### CODE REVIEW

**Review by tbro — COMMENTED** [review]
Looks good. I left a couple of comments.

File: types/src/v0/impls/block/full_payload/payload.rs [hunk 1]
```diff
@@ -61,9 +79,12 @@ impl Payload {
...
     > {
         // accounting for block byte length limit
-        let max_block_byte_len: usize = u64::from(chain_config.max_block_size)
-            .try_into()
-            .map_err(|_| <Self as BlockPayload<SeqTypes>>::Error::BlockBuilding)?;
+        let max_block_byte_len: usize =
+            u64::from(chain_config.max_block_size)
+                .try_into()
+                .map_err(|_| {
+                    <Self as BlockPayload<SeqTypes>>::Error::Custom("cast u64 to usize".to_string())
```
[the remark this answers is not in the document]
Comment (imabdulbasit) [reply types/src/v0/impls/block/full_payload/payload.rs:86]: Yeah, I agree. I'll remove custom and panic here

**Comment by tbro** [comment]
I think this addresses #1539, and Closes #1681 and  #1699. Is that right?

**Comment by imabdulbasit** [comment]
> I think this addresses #1539, and Closes #1681 and #1699. Is that right?

So it doesn't completely remove the snafu , as the query service is using it and we are extending the availability endpoint. However, I have replaced it with this_error everywhere else

**Comment by imabdulbasit** [comment]
> I think this addresses #1539, and Closes #1681 and #1699. Is that right?

Yes

**Comment by tbro** [comment]
> So it doesn't completely remove the snafu , as the query service is using it and we are extending the availability endpoint. However, I have replaced it with this_error everywhere else

Ok so #1539 will remain open for now

**Review by tbro — APPROVED** [review]

**Pull request merged**